api/src/main/java/org/openmrs/module/fgh/mpi/SnapshotEventProcessor.java

```diff
@@ -128,27 +142,79 @@ public void process(DatabaseEvent event) {
 				});
 				
 				if (!fhirPatients.isEmpty()) {
-					Map<String, Object> fhirBundle = new HashMap(3);
-					fhirBundle.put(MpiConstants.FIELD_RESOURCE_TYPE, MpiConstants.BUNDLE);
-					fhirBundle.put(MpiConstants.FIELD_TYPE, MpiConstants.BATCH);
-					fhirBundle.put(MpiConstants.FIELD_ENTRY, fhirPatients);
+					int successPatientCount = 0;
 					
-					List<Object> response = mpiHttpClient.submitBundle(mapper.writeValueAsString(fhirBundle));
-					//The response is a list of 2 MPI identifiers for each patient
-					int successPatientCount = response.size() / 2;
-					if (fhirPatients.size() == successPatientCount) {
-						if (log.isDebugEnabled()) {
-							log.debug("All patients in the batch were successfully processed by the MPI");
+					if (mpiContext.getMpiSystem().isOpenCr()) {
+						Map<String, Object> fhirBundle = new HashMap(3);
+						fhirBundle.put(MpiConstants.FIELD_RESOURCE_TYPE, MpiConstants.BUNDLE);
+						fhirBundle.put(MpiConstants.FIELD_TYPE, MpiConstants.BATCH);
+						fhirBundle.put(MpiConstants.FIELD_ENTRY, fhirPatients);
+						
+						List<Object> response = mpiHttpClient.submitBundle("fhir", mapper.writeValueAsString(fhirBundle),
+						    List.class);
+						
+						successPatientCount = response.size() / 2;
+						
+						if (fhirPatients.size() == successPatientCount) {
+							if (log.isDebugEnabled()) {
+								log.debug("All patients in the batch were successfully processed by the MPI");
+							}
+							
+							successCount.addAndGet(fhirPatients.size());
+						} else {
+							//TODO Loop through all patients in the batch and check which records were problematic
+							throw new APIException((fhirPatients.size() - successPatientCount)
+							        + " patient(s) in the batch were not successfully processed by the MPI");
 						}
 						
-						successCount.addAndGet(fhirPatients.size());
```

Comment:
Do you realize this line of code is not executing for Sante? Please make sure it is executed for Sante  also.